src/App/App.js

```diff
@@ -57,71 +33,17 @@ class App extends Component {
 
     this.setState({ ...this.state, error: 'Looks like we where unable to find that title.' });
   }
-
-  viewMovie = (id) => {
-    const movie = this.findMovieById(id);
-    if(movie) {
-      this.setState({
-            ...this.state, 
-            isMovieView: true,
-            error: '',
-            movie: {
-            id: movie.id,
-            title: movie.title,
-            posterPath: movie['poster_path'],
-            backdropPath: movie['backdrop_path'],
-            releaseDate: movie['release_date'],
-            overview: movie.overview,
-            genres: movie.genres,
-            budget: movie.budget,
-            revenue: movie.revenue,
-            runtime: movie.runtime,
-            tagline: movie.tagline,
-            averageRating: movie['average_rating']
-          }})
-          
-          return;
-    }
-
-    getMovie(id)
-      .then(response => response.json())
-      .then(data => {
-        this.setState({...this.state, 
-          isMovieView: true,
-          fetchedMovies: [...this.state.fetchedMovies, data.movie],
-          error: '',
-          movie: {
-          id: data.movie.id,
-          title: data.movie.title,
-          posterPath: data.movie['poster_path'],
-          backdropPath: data.movie['backdrop_path'],
-          releaseDate: data.movie['release_date'],
-          overview: data.movie.overview,
-          genres: data.movie.genres,
-          budget: data.movie.budget,
-          revenue: data.movie.revenue,
-          runtime: data.movie.runtime,
-          tagline: data.movie.tagline,
-          averageRating: data.movie['average_rating']
-        }})
-      })
-      .catch(err =>{
-        console.log(err)
-        this.setState({ ...this.state, error: 'Looks like something went wrong.' })
-      })
-  }
  
   render() {
     return (
       <div className="App">
         <Navbar 
           view={this.state.isMovieView} 
-          goHome={ this.goHome }
           findMovieByTitle={this.findMovieByTitle}
         />
         {this.state.error && <h3 className='error'>{this.state.error}</h3>}
-        {this.state.isMovieView || <AllMovies movies={this.state.movies} switchView={this.viewMovie}/>}
-        {this.state.isMovieView && <Movie movieDetails={this.state.movie}/>}
+          <Route exact path='/' render={() => <AllMovies movies={this.state.movies}/>}/>
+          <Route exact path='/:id' render={({ match }) => <Movie id={match.params.id} />}/>
```

Comment:
used the `match` to collect the movie `id`, when the Movie component loads it uses the prop (id) passed in to fetch the movie